amethyst_network/src/systems/network_client.rs

```diff
@@ -0,0 +1,119 @@
+use std::clone::Clone;
+
+use shrev::*;
+
+use resources::*;
+use systems::NetworkBase;
+use std::marker::PhantomData;
+
+use serde::Serialize;
+use serde::de::DeserializeOwned;
+
+/// The System managing the client's network state and connections.
+/// The T generic parameter corresponds to the network event enum type.
+pub struct NetClientSystem<T> where T:Send+Sync{
+    /// The network socket
+    pub socket:UdpSocket,
+    /// The server that we are (possibly) connected or connecting to.
+    pub connection:Option<NetConnection>,//Will handle a single connection for now, as it is simple to manage and corresponds to most use cases.
+    net_event_types:PhantomData<T>,
+}
+
+//TODO: add Unchecked Event type list. Those events will be let pass the client connected filter (Example: NetEvent::Connect).
+//TODO: add different Filters that can be added on demand, to filter the event before they reach other systems.
+impl<T> NetClientSystem<T> where T:Send+Sync+Serialize+Clone+DeserializeOwned+BaseNetEvent<T>+'static{
+    /// Creates a NetClientSystem and binds the Socket on the ip and port added in parameters.
+    pub fn new(ip:&str,port:u16)->Result<NetClientSystem<T>,Error>{
+        let socket = UdpSocket::bind(SocketAddr::new(IpAddr::from_str(ip).expect("Unreadable input IP"),port))?;
+        socket.set_nonblocking(true)?;
+        Ok(
+            NetClientSystem{
+                socket,
+                connection:None,
+                net_event_types:PhantomData,
+            }
+        )
+    }
+    /// Connects to a remote server
+    pub fn connect(&mut self,target:SocketAddr){
+        println!("Sending connection request to remote {:?}",target);
+        let conn = NetConnection{
+            target,
+            state:ConnectionState::Connecting,
+        };
+        self.send_event(&T::base_to_custom(NetEvent::Connect),&conn,&self.socket);
+        self.connection = Some(
+            conn,
+        );
+    }
+}
+
+impl<T> NetworkBase<T> for NetClientSystem<T> where T:Send+Sync+Serialize+Clone+DeserializeOwned+BaseNetEvent<T>+'static{}
+
+impl<'a, T> System<'a> for NetClientSystem<T> where T:Send+Sync+Serialize+Clone+DeserializeOwned+BaseNetEvent<T>+'static{
+    type SystemData = (
+        FetchMut<'a, EventChannel<NetOwnedEvent<T>>>,
+    );
+    //omg unreadable plz enjoy code owo
+    fn run(&mut self, (mut events,): Self::SystemData) {
+        let mut buf = [0; 2048];
+        loop {
+            match self.socket.recv_from(&mut buf) {
+                //Data received
+                Ok((amt, src)) => {
+                    //Are we connected to anything?
+                    if self.connection.is_some(){
+                        //Was it sent by connected server, and are we still connected to it?
+                        if src == self.connection.as_ref().unwrap().target && (self.connection.as_ref().unwrap().state == ConnectionState::Connected || self.connection.as_ref().unwrap().state == ConnectionState::Connecting){
```

Comment:
You can probably bundle these if statements into a function that returns `bool`.

Reply:
this will get replaced by `Filter`s in a future PR as written in the Networking RFC ;)